sdks/python/apache_beam/io/gcp/bigquery.py

```diff
@@ -496,6 +505,233 @@ def reader(self, test_bigquery_client=None):
         kms_key=self.kms_key)
 
 
+SchemaFields = collections.namedtuple('SchemaFields', 'fields mode name type')
+
+
+def _to_bool(value):
+  return value == 'true'
+
+
+def _to_decimal(value):
+  return decimal.Decimal(value)
+
+
+def _to_bytes(value):
+  """Converts value from str to bytes on Python 3.x. Does nothing on
+  Python 2.7."""
+  return value.encode('utf-8')
+
+
+class _BigQueryRowCoder(coders.Coder):
```

Comment:
I feel silly - but didn't we have a coder for this? I see that `TableRowJsonCoder` does not do the full tablerow to dict conversion... does it make sense to extend it? Or not really?
[1] https://github.com/apache/beam/blob/12d07745835e1b9c1e824b83beeeadf63ab4b234/sdks/python/apache_beam/io/gcp/bigquery.py#L312-L349

Reply:
Looks interesting. If I combine this coder with this function[1], maybe that would be the solution. I'll investigate this further — first I need to solve the problem that `bigquery.TableSchema` is unpickable.

[1] https://github.com/apache/beam/blob/03f780c7329e0eca692baef44874056b7d263303/sdks/python/apache_beam/io/gcp/bigquery_tools.py#L892-L920

Reply:
I looked at this and, sadly, it doesn't work in practice. For each field `TableRowJsonCoder` returns `JsonValue` instance which doesn't contain the name of the given field. This makes finding an appropriate field type and matching a conversion function quite hard.

That's why I decided to stay with my own coder.